File: STIF_Algo/HMM_package/pgms/sdevn_all_files.pl
#!/usr/bin/perl

$perl_script = $ARGV[0];
print $perl_script;

opendir DIR,"$ARGV[1]";
@dir = <DIR>;


while ($i = readdir(DIR))
{
        if($i =~ /(\w+)_full\d*/)
        {
                $file_name = $1;
                print "SS :: $i :: $var1\n";
                $cmd1 = "perl $perl_script\/sdevn.pl $ARGV[1]\/$i \> $ARGV[1]\/$file_name\.sd";
                #print "$cmd1\n";
                system($cmd1);

		$n++;
	}
}
print "CNT :: $n\n";
closedir(DIR);

File: STIF_Algo/HMM_package/pgms/sdevn_all_files.py
#!/usr/bin/env python3

import os
import sys
import re

# Get Perl script path from command-line argument
perl_script = sys.argv[1]
print(perl_script)

# Open directory given in second command-line argument
dir_path = sys.argv[2]
dir_list = os.listdir(dir_path)

n = 0  # Counter for processed files

for filename in dir_list:
    if re.match(r'(\w+)_full\d*', filename):
        file_name = re.match(r'(\w+)_full\d*', filename).group(1)
        print(f"SS :: {filename} :: {file_name}\n")
        cmd1 = f"perl {perl_script}/sdevn.pl {dir_path}/{filename} > {dir_path}/{file_name}.sd"
        #print(f"{cmd1}\n")
        os.system(cmd1)
        n += 1

print(f"CNT :: {n}\n")